[Improvement]  In GravitinoErrorCode.java errorCode in not Immutable

### What would you like to be improved?

GravitinoErrorCode.java:39: warning: [ImmutableEnumChecker] enums should be immutable: 'GravitinoErrorCode' has field 'errorCode' of type 'io.trino.spi.ErrorCode', the declaration of type 'io.trino.spi.ErrorCode' is not annotated with @com.google.errorprone.annotations.Immutable
  private final ErrorCode errorCode;

See https://errorprone.info/bugpattern/ImmutableEnumChecker

### How should we improve?

_No response_

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -252,6 +252,7 @@ subprojects {
         "IdentityBinaryExpression",
         "IdentityHashMapBoxing",
         "Immutable",
+        "ImmutableEnumChecker",
         "Incomparable",
         "IncompatibleArgumentType",
         "IndexOfChar",

--- a/trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoErrorCode.java
+++ b/trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoErrorCode.java
@@ -36,6 +36,7 @@ public enum GravitinoErrorCode implements ErrorCodeSupplier {
   GRAVITINO_OPERATION_FAILED(22, EXTERNAL),
   ;
 
+  @SuppressWarnings("ImmutableEnumChecker")
   private final ErrorCode errorCode;
 
   GravitinoErrorCode(int code, ErrorType type) {

--- a/trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoErrorCode.java
+++ b/trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoErrorCode.java
@@ -36,6 +36,7 @@ public enum GravitinoErrorCode implements ErrorCodeSupplier {
   GRAVITINO_OPERATION_FAILED(22, EXTERNAL),
   ;
 
+  // suppress ImmutableEnumChecker because ErrorCode is outside the project.
   @SuppressWarnings("ImmutableEnumChecker")
   private final ErrorCode errorCode;
 

